src/terms/nonlocal.jl

```diff
@@ -30,11 +30,9 @@ end
 
     E = zero(T)
     for (ik, k) in enumerate(basis.kpoints)
-        for iband = 1:size(ψ[1], 2)
-            ψnk = @views ψ[ik][:, iband]
-            Pψnk = term.ops[ik].P' * ψnk
-            E += basis.kweights[ik] * occ[ik][iband] * real(dot(Pψnk, term.ops[ik].D * Pψnk))
-        end
+        Pψ = term.ops[ik].P' * ψ[ik] # nproj x nband
+        band_enes = dropdims(sum(conj.(Pψ) .* (term.ops[ik].D*Pψ), dims=1), dims=1)
```

Comment:
Space around `*`.

Reply:
I don't think you need the `dropdims` because in the next line you could just do `.* occ[ik]'`. Also I'd put the `real` in the line computing the `band_enes` (and I would call it `PDPψ` not `band_enes`)

Reply:
Why? It is the band energies

Reply:
dropdims is just simpler because otherwise you have to think about sizes

Reply:
But now I wonder why you don't play the same trick with the real as above at this instance?

Reply:
very good point!